datasource/netcdf_groundwater.py

```diff
@@ -75,48 +152,120 @@ def line_type_of(self, line_idx):
         return self.get_timestamps()
 
+    # def get_timeseries(
+    #         self, object_type, object_id, nc_variable, fill_value=None):
+    #     return self.temporary_get_timeseries_impl(
+    #         object_type, object_id, nc_variable, fill_value)
+
+    def get_timeseries_schematisation_layer(
+            self, object_type, object_id, nc_variable, timeseries=None,
+            only=None, data=None):
+
+        """
+        -   just using a random v2_bergermeer nc and h5 file we see that:
+        In [1]: gr.lines.filter(content_type='v2_channel').content_pk
+        Out[1]: array([1, 1, 2, ...,  760,  851, 1090]
+        In [2]: len(set(gr.lines.filter(content_type='v2_channel').content_pk))
+        Out[2]: 2346
+        In [3]: len(set(gr.lines.filter(content_type='v2_channel').content_pk))
+        Out[3]: 1175
+        -   The v2_channel QgisVectorLayer contains id 1 till 666668899 (
+            1175 rows), so we can use content_pk to retrieve a timeseries
+        -   to get a timeseries using threedigridadmin, we need to call e.g.
+            gr.lines.filter(content_type='v2_channel', content_pk=1).
+            timeseries(indexes=slice(None)).only('q').data['q']
+        -   this will return multiple timeseries (since a v2_channel can be
+            splitted up in multiple flowlines
+        -   slice(None) means we get all timesteps.
+        -   get_timeseries_schematisation_layer() gets strings as arguments
+            that we need to parse as atrributes
+        """
+
+        # one example for v2_weir_view
+        # gr.lines
+        # determine wheter v2_weir_view is a lines, nodes or pumps
+        first = getattr(self.gr, get_gr_type(object_type))
+        # gr.lines.filter(content_type='v2_weir', content_pk=1)
+        # we need to get de gridadminname of v2_weir_view (=v2_weir)
+        second = first.filter(content_type=get_gr_name(object_type),
+                              content_pk=object_id)
+        # gr.lines.filter(content_type='v2_channel',content_pk=1).timeseries(
+        # indexes=slice(None))
+        third = second.timeseries(indexes=slice(None))
+        # gr.lines.filter(content_type='v2_channel',content_pk=1).timeseries(
+        # indexes=slice(None)).only('q')
+        fourth = third.only(nc_variable)
+        # gr.lines.filter(content_type='v2_channel',content_pk=1).timeseries(
+        # indexes=slice(None)).only('q').data
+        fifth = fourth.data
+        # fifth is an OrderedDict:
+        # OrderedDict([('q', array([  [ 0.00000000e+00,  0.00000000e+00],
+        #                             [-3.27184124e-10, -7.85903914e-12],
+        # to only get the array we use fifth['q']
+
+        # for now, we just take only the timeseries of the first flowline (
+        # when multiple flowlines are returned
+        # TODO
+        # return all timeseries are do an avg, min, max
+        # e.g. self.vals = np.mean(fifth[nc_variable], axis=1)
+        self.vals = fifth[nc_variable][:, 0]
```

Comment:
No, don't do this. Please program using function with input and output values: https://upload.wikimedia.org/wikipedia/commons/thumb/3/3b/Function_machine2.svg/220px-Function_machine2.svg.png